### PR TITLE
feat: add edit/delete features

### PR DESCRIPTION
**Pull request by nidalhjab**
- Add edit/delete buttons to the locations table with their functionalities.
- Add LoginModal and LocationModal with custom styles.
- Edit the table components to be reused in the modal.

### CODE REVIEW

File: packages/client/public/assets/locales/en/translation.json [hunk 1]
```diff
@@ -56,6 +56,11 @@
 			"message": "Would you like to set-up favorite locations?",
 			"accept": "Yes",
 			"decline": "Cancel"
+		},
+		"location": {
+			"message": "Add the updated location by clicking on the map",
+			"accept": "Save",
+			"decline": "Cancle"
```
Comment (mohannadjamal) [inline packages/client/public/assets/locales/en/translation.json:63]: Cancel typo

File: packages/client/src/common/components/Modal/LocationModal.tsx [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+import { Box, Button, Divider, TextField, Typography } from "@mui/material";
+import {Map} from '../Map/Map';
+import Table from "../Table/Table";
+import { useNavigate } from "react-router-dom";
+import { trainingClient } from "../../api/trainingClient";
+import { useMe } from "../../hooks/useMe.hook";
+import { Data } from "@react-google-maps/api";
+import { TextInput } from "../TextInput/TextInput";
+
+const ModalBox = styled(Box)({
```
Comment (mohannadjamal) [inline packages/client/src/common/components/Modal/LocationModal.tsx:14]: Preferably styled constants should be in LocationModal.styles.ts

**Review by mohannadjamal — CHANGES_REQUESTED** [review]

File: packages/client/src/common/components/Map/Map.tsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
+import { Box } from "@mui/material";
+import { GoogleMap, Marker, useJsApiLoader } from "@react-google-maps/api";
+import { useState, useCallback } from "react";
+interface MapProps{
+	position:{
+    lat:number,
+    lng:number
+	},
+	modalCallback:(lat:number,lng:number)=>void
+}
```
Comment (hamzahmurrar) [inline packages/client/src/common/components/Map/Map.tsx:10]: fix indentation

File: packages/client/src/common/components/Map/Map.tsx [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+		(map: google.maps.Map) => {
+			const bounds = new window.google.maps.LatLngBounds(position);
+			map.fitBounds(bounds);
+			setMap(map);
+		},
+		[position]
+	);
+	
+	const handleNewPosition = (event?:google.maps.LatLngLiteral)=>{
+		if(event){
```
Comment (hamzahmurrar) [inline packages/client/src/common/components/Map/Map.tsx:28]: `if (event) {`

**Review by mohannadjamal — APPROVED** [review]

**Comment by nidalhjab** [comment]
> fix conflicts please

already fixed @hamzahmurrar

**Review by hamzahmurrar — APPROVED** [review]

**Pull request merged**